java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/common/CloseableRMMAllocation.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.cuvs.internal.common;
+
+import static com.nvidia.cuvs.internal.common.Util.checkCuVSError;
+import static com.nvidia.cuvs.internal.panama.headers_h.cuvsRMMFree;
+
+import java.lang.foreign.MemorySegment;
+
+/**
+ * A closeable handle for RMM allocations that can be used with try-with-resources.
+ * This class manages the lifecycle of a memory allocation made with RMM (RAPIDS Memory Manager).
+ * It ensures that the allocated memory is properly released when no longer needed.
+ */
+public class CloseableRMMAllocation implements CloseableHandle {
+
+  private final long cuvsResourceHandle;
+  private final long numBytes;
+  private MemorySegment pointer;
+
+  public CloseableRMMAllocation(long cuvsResourceHandle, long numBytes, MemorySegment pointer) {
+    this.cuvsResourceHandle = cuvsResourceHandle;
+    this.numBytes = numBytes;
+    this.pointer = pointer;
+  }
+
+  @Override
+  public MemorySegment handle() {
+    return pointer;
+  }
+
+  public MemorySegment release() {
```

Comment:
See above comment: I think it's better to not have "release", we can just avoid to use try (and avoid calling close) when not needed.

Reply:
[Counterpoint.](https://github.com/rapidsai/cuvs/pull/1215#discussion_r2255077242)

Reply:
Closing this thread.  We can carry on this discussion on the [main thread](https://github.com/rapidsai/cuvs/pull/1215#discussion_r2253915198).